src/data/init/init-abilities.ts

```diff
@@ -890,11 +890,13 @@ export function initAbilities() {
       .attr(PostBattleInitFormChangeAbAttr, () => 0)
       .attr(PostSummonFormChangeAbAttr, (p) => (p.formIndex % 7) + (p.getHpRatio() <= 0.5 ? 7 : 0))
       .attr(PostTurnFormChangeAbAttr, (p) => (p.formIndex % 7) + (p.getHpRatio() <= 0.5 ? 7 : 0))
+      .conditionalAttr(p => p.formIndex !== 7, StatusEffectImmunityAbAttr)
+      .conditionalAttr(p => p.formIndex !== 7, BattlerTagImmunityAbAttr, BattlerTagType.DROWSY)
```

Comment:
```suggestion
      .conditionalAttr(p => p.formIndex < 7, StatusEffectImmunityAbAttr)
      .conditionalAttr(p => p.formIndex < 7, BattlerTagImmunityAbAttr, BattlerTagType.DROWSY)
```
It might be a good idea to make sure the source is Minior here, too.

Reply:
There is no species check for Pokémon specific abilities like this one

[Power Construct](https://github.com/Despair-Games/poketernity/blob/beta/src/data/init/init-abilities.ts#L986) :
```
      .conditionalAttr(
        (pokemon) => pokemon.formIndex === 2 || pokemon.formIndex === 4,
        PostBattleInitFormChangeAbAttr,
        () => 2,
      )
```

[Disguise](https://github.com/Despair-Games/poketernity/blob/beta/src/data/init/init-abilities.ts#L951) :
```
      .conditionalAttr(
        (pokemon) => pokemon.formIndex === 0,
        PostSummonAddBattlerTagAbAttr,
        BattlerTagType.DISGUISE,
        0,
        false,
      )
```

[Ice Face](https://github.com/Despair-Games/poketernity/blob/beta/src/data/init/init-abilities.ts#L1189) :
```
      .conditionalAttr(
        (pokemon) => pokemon.formIndex === 0,
        PostSummonAddBattlerTagAbAttr,
        BattlerTagType.ICE_FACE,
        0,
        false,
      )
```

So either I don't understand what you mean by checking that the _source_ is Minior, or the species check might be worth to add on a few other abilities

Reply:
I meant the "source" as in the Pokemon with the ability. Basing the logic on form indexes only works under the assumption that the source is Minior. I figure it'd be better to explicitly make sure that's the case.

Historically, we haven't enforced species locks on abilties mainly to make them compatible with fusions. Fusions don't exist anymore, though, so I think it'd be safer to add the species checks